From: J. R. Okajima <hooanon05@yahoo.co.jp>
Date: Sun Mar 10 01:25:13 2019 +0900
Subject: aufs5.0 standalone patch
Origin: https://github.com/sfjro/aufs5-standalone/tree/06b60dd6e4ffa5e1cdf199a578ba1d3acac6821d
Bug-Debian: https://bugs.debian.org/541828

Patch headers added by debian/bin/genpatch-aufs

SPDX-License-Identifier: GPL-2.0
aufs5.0 standalone patch

Index: debian-kernel/fs/dcache.c
===================================================================
--- debian-kernel.orig/fs/dcache.c
+++ debian-kernel/fs/dcache.c
@@ -1352,6 +1352,7 @@ rename_retry:
 	seq = 1;
 	goto again;
 }
+EXPORT_SYMBOL_GPL(d_walk);
 
 struct check_mount {
 	struct vfsmount *mnt;
@@ -2845,6 +2846,7 @@ void d_exchange(struct dentry *dentry1,
 
 	write_sequnlock(&rename_lock);
 }
+EXPORT_SYMBOL_GPL(d_exchange);
 
 /**
  * d_ancestor - search for an ancestor
Index: debian-kernel/fs/exec.c
===================================================================
--- debian-kernel.orig/fs/exec.c
+++ debian-kernel/fs/exec.c
@@ -109,6 +109,7 @@ bool path_noexec(const struct path *path
 	return (path->mnt->mnt_flags & MNT_NOEXEC) ||
 	       (path->mnt->mnt_sb->s_iflags & SB_I_NOEXEC);
 }
+EXPORT_SYMBOL_GPL(path_noexec);
 
 #ifdef CONFIG_USELIB
 /*
Index: debian-kernel/fs/fcntl.c
===================================================================
--- debian-kernel.orig/fs/fcntl.c
+++ debian-kernel/fs/fcntl.c
@@ -85,6 +85,7 @@ int setfl(int fd, struct file * filp, un
  out:
 	return error;
 }
+EXPORT_SYMBOL_GPL(setfl);
 
 static void f_modown(struct file *filp, struct pid *pid, enum pid_type type,
                      int force)
Index: debian-kernel/fs/file_table.c
===================================================================
--- debian-kernel.orig/fs/file_table.c
+++ debian-kernel/fs/file_table.c
@@ -161,6 +161,7 @@ over:
 	}
 	return ERR_PTR(-ENFILE);
 }
+EXPORT_SYMBOL_GPL(alloc_empty_file);
 
 /*
  * Variant of alloc_empty_file() that doesn't check and modify nr_files.
@@ -323,6 +324,7 @@ void flush_delayed_fput(void)
 {
 	delayed_fput(NULL);
 }
+EXPORT_SYMBOL_GPL(flush_delayed_fput);
 
 static DECLARE_DELAYED_WORK(delayed_fput_work, delayed_fput);
 
@@ -370,6 +372,7 @@ void __fput_sync(struct file *file)
 }
 
 EXPORT_SYMBOL(fput);
+EXPORT_SYMBOL_GPL(__fput_sync);
 
 void __init files_init(void)
 {
Index: debian-kernel/fs/inode.c
===================================================================
--- debian-kernel.orig/fs/inode.c
+++ debian-kernel/fs/inode.c
@@ -1666,6 +1666,7 @@ int update_time(struct inode *inode, str
 
 	return update_time(inode, time, flags);
 }
+EXPORT_SYMBOL_GPL(update_time);
 
 /**
  *	touch_atime	-	update the access time
Index: debian-kernel/fs/namespace.c
===================================================================
--- debian-kernel.orig/fs/namespace.c
+++ debian-kernel/fs/namespace.c
@@ -435,6 +435,7 @@ void __mnt_drop_write(struct vfsmount *m
 	mnt_dec_writers(real_mount(mnt));
 	preempt_enable();
 }
+EXPORT_SYMBOL_GPL(__mnt_drop_write);
 
 /**
  * mnt_drop_write - give up write access to a mount
@@ -774,6 +775,7 @@ int is_current_mnt_ns(struct vfsmount *m
 {
 	return check_mnt(real_mount(mnt));
 }
+EXPORT_SYMBOL_GPL(is_current_mnt_ns);
 
 /*
  * vfsmount lock must be held for write
@@ -1846,6 +1848,7 @@ void drop_collected_mounts(struct vfsmou
 	unlock_mount_hash();
 	namespace_unlock();
 }
+EXPORT_SYMBOL_GPL(iterate_mounts);
 
 /**
  * clone_private_mount - create a private clone of a path
Index: debian-kernel/fs/notify/group.c
===================================================================
--- debian-kernel.orig/fs/notify/group.c
+++ debian-kernel/fs/notify/group.c
@@ -112,6 +112,7 @@ void fsnotify_get_group(struct fsnotify_
 {
 	refcount_inc(&group->refcnt);
 }
+EXPORT_SYMBOL_GPL(fsnotify_get_group);
 
 /*
  * Drop a reference to a group.  Free it if it's through.
@@ -121,6 +122,7 @@ void fsnotify_put_group(struct fsnotify_
 	if (refcount_dec_and_test(&group->refcnt))
 		fsnotify_final_destroy_group(group);
 }
+EXPORT_SYMBOL_GPL(fsnotify_put_group);
 
 /*
  * Create a new fsnotify_group and hold a reference for the group returned.
@@ -150,6 +152,7 @@ struct fsnotify_group *fsnotify_alloc_gr
 
 	return group;
 }
+EXPORT_SYMBOL_GPL(fsnotify_alloc_group);
 
 int fsnotify_fasync(int fd, struct file *file, int on)
 {
Index: debian-kernel/fs/notify/mark.c
===================================================================
--- debian-kernel.orig/fs/notify/mark.c
+++ debian-kernel/fs/notify/mark.c
@@ -289,6 +289,7 @@ void fsnotify_put_mark(struct fsnotify_m
 	queue_delayed_work(system_unbound_wq, &reaper_work,
 			   FSNOTIFY_REAPER_DELAY);
 }
+EXPORT_SYMBOL_GPL(fsnotify_put_mark);
 
 /*
  * Get mark reference when we found the mark via lockless traversal of object
@@ -443,6 +444,7 @@ void fsnotify_destroy_mark(struct fsnoti
 	mutex_unlock(&group->mark_mutex);
 	fsnotify_free_mark(mark);
 }
+EXPORT_SYMBOL_GPL(fsnotify_destroy_mark);
 
 /*
  * Sorting function for lists of fsnotify marks.
@@ -685,6 +687,7 @@ int fsnotify_add_mark(struct fsnotify_ma
 	mutex_unlock(&group->mark_mutex);
 	return ret;
 }
+EXPORT_SYMBOL_GPL(fsnotify_add_mark);
 
 /*
  * Given a list of marks, find the mark associated with given group. If found
@@ -809,6 +812,7 @@ void fsnotify_init_mark(struct fsnotify_
 	mark->group = group;
 	WRITE_ONCE(mark->connector, NULL);
 }
+EXPORT_SYMBOL_GPL(fsnotify_init_mark);
 
 /*
  * Destroy all marks in destroy_list, waits for SRCU period to finish before
Index: debian-kernel/fs/open.c
===================================================================
--- debian-kernel.orig/fs/open.c
+++ debian-kernel/fs/open.c
@@ -64,6 +64,7 @@ int do_truncate(struct dentry *dentry, l
 	inode_unlock(dentry->d_inode);
 	return ret;
 }
+EXPORT_SYMBOL_GPL(do_truncate);
 
 long vfs_truncate(const struct path *path, loff_t length)
 {
Index: debian-kernel/fs/read_write.c
===================================================================
--- debian-kernel.orig/fs/read_write.c
+++ debian-kernel/fs/read_write.c
@@ -459,6 +459,7 @@ ssize_t vfs_read(struct file *file, char
 
 	return ret;
 }
+EXPORT_SYMBOL_GPL(vfs_read);
 
 static ssize_t new_sync_write(struct file *filp, const char __user *buf, size_t len, loff_t *ppos)
 {
@@ -499,6 +500,7 @@ vfs_readf_t vfs_readf(struct file *file)
 		return new_sync_read;
 	return ERR_PTR(-ENOSYS);
 }
+EXPORT_SYMBOL_GPL(vfs_readf);
 
 vfs_writef_t vfs_writef(struct file *file)
 {
@@ -510,6 +512,7 @@ vfs_writef_t vfs_writef(struct file *fil
 		return new_sync_write;
 	return ERR_PTR(-ENOSYS);
 }
+EXPORT_SYMBOL_GPL(vfs_writef);
 
 ssize_t __kernel_write(struct file *file, const void *buf, size_t count, loff_t *pos)
 {
@@ -579,6 +582,7 @@ ssize_t vfs_write(struct file *file, con
 
 	return ret;
 }
+EXPORT_SYMBOL_GPL(vfs_write);
 
 static inline loff_t file_pos_read(struct file *file)
 {
Index: debian-kernel/fs/splice.c
===================================================================
--- debian-kernel.orig/fs/splice.c
+++ debian-kernel/fs/splice.c
@@ -846,6 +846,7 @@ long do_splice_from(struct pipe_inode_in
 
 	return splice_write(pipe, out, ppos, len, flags);
 }
+EXPORT_SYMBOL_GPL(do_splice_from);
 
 /*
  * Attempt to initiate a splice from a file to a pipe.
@@ -875,6 +876,7 @@ long do_splice_to(struct file *in, loff_
 
 	return splice_read(in, ppos, pipe, len, flags);
 }
+EXPORT_SYMBOL_GPL(do_splice_to);
 
 /**
  * splice_direct_to_actor - splices data directly between two non-pipes
Index: debian-kernel/fs/sync.c
===================================================================
--- debian-kernel.orig/fs/sync.c
+++ debian-kernel/fs/sync.c
@@ -39,6 +39,7 @@ int __sync_filesystem(struct super_block
 		sb->s_op->sync_fs(sb, wait);
 	return __sync_blockdev(sb->s_bdev, wait);
 }
+EXPORT_SYMBOL_GPL(__sync_filesystem);
 
 /*
  * Write out and wait upon all dirty data associated with this
Index: debian-kernel/fs/xattr.c
===================================================================
--- debian-kernel.orig/fs/xattr.c
+++ debian-kernel/fs/xattr.c
@@ -295,6 +295,7 @@ vfs_getxattr_alloc(struct dentry *dentry
 	*xattr_value = value;
 	return error;
 }
+EXPORT_SYMBOL_GPL(vfs_getxattr_alloc);
 
 ssize_t
 __vfs_getxattr(struct dentry *dentry, struct inode *inode, const char *name,
Index: debian-kernel/kernel/locking/lockdep.c
===================================================================
--- debian-kernel.orig/kernel/locking/lockdep.c
+++ debian-kernel/kernel/locking/lockdep.c
@@ -162,6 +162,7 @@ inline struct lock_class *lockdep_hlock_
 	}
 	return lock_classes + hlock->class_idx - 1;
 }
+EXPORT_SYMBOL_GPL(lockdep_hlock_class);
 #define hlock_class(hlock) lockdep_hlock_class(hlock)
 
 #ifdef CONFIG_LOCK_STAT
Index: debian-kernel/kernel/task_work.c
===================================================================
--- debian-kernel.orig/kernel/task_work.c
+++ debian-kernel/kernel/task_work.c
@@ -117,3 +117,4 @@ void task_work_run(void)
 		} while (work);
 	}
 }
+EXPORT_SYMBOL_GPL(task_work_run);
Index: debian-kernel/security/device_cgroup.c
===================================================================
--- debian-kernel.orig/security/device_cgroup.c
+++ debian-kernel/security/device_cgroup.c
@@ -824,3 +824,4 @@ int __devcgroup_check_permission(short t
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(__devcgroup_check_permission);
Index: debian-kernel/security/security.c
===================================================================
--- debian-kernel.orig/security/security.c
+++ debian-kernel/security/security.c
@@ -996,6 +996,7 @@ int security_path_rmdir(const struct pat
 		return 0;
 	return call_int_hook(path_rmdir, 0, dir, dentry);
 }
+EXPORT_SYMBOL_GPL(security_path_rmdir);
 
 int security_path_unlink(const struct path *dir, struct dentry *dentry)
 {
@@ -1012,6 +1013,7 @@ int security_path_symlink(const struct p
 		return 0;
 	return call_int_hook(path_symlink, 0, dir, dentry, old_name);
 }
+EXPORT_SYMBOL_GPL(security_path_symlink);
 
 int security_path_link(struct dentry *old_dentry, const struct path *new_dir,
 		       struct dentry *new_dentry)
@@ -1020,6 +1022,7 @@ int security_path_link(struct dentry *ol
 		return 0;
 	return call_int_hook(path_link, 0, old_dentry, new_dir, new_dentry);
 }
+EXPORT_SYMBOL_GPL(security_path_link);
 
 int security_path_rename(const struct path *old_dir, struct dentry *old_dentry,
 			 const struct path *new_dir, struct dentry *new_dentry,
@@ -1047,6 +1050,7 @@ int security_path_truncate(const struct
 		return 0;
 	return call_int_hook(path_truncate, 0, path);
 }
+EXPORT_SYMBOL_GPL(security_path_truncate);
 
 int security_path_chmod(const struct path *path, umode_t mode)
 {
@@ -1054,6 +1058,7 @@ int security_path_chmod(const struct pat
 		return 0;
 	return call_int_hook(path_chmod, 0, path, mode);
 }
+EXPORT_SYMBOL_GPL(security_path_chmod);
 
 int security_path_chown(const struct path *path, kuid_t uid, kgid_t gid)
 {
@@ -1061,6 +1066,7 @@ int security_path_chown(const struct pat
 		return 0;
 	return call_int_hook(path_chown, 0, path, uid, gid);
 }
+EXPORT_SYMBOL_GPL(security_path_chown);
 
 int security_path_chroot(const struct path *path)
 {
@@ -1161,6 +1167,7 @@ int security_inode_permission(struct ino
 		return 0;
 	return call_int_hook(inode_permission, 0, inode, mask);
 }
+EXPORT_SYMBOL_GPL(security_inode_permission);
 
 int security_inode_setattr(struct dentry *dentry, struct iattr *attr)
 {
@@ -1332,6 +1339,7 @@ int security_file_permission(struct file
 
 	return fsnotify_perm(file, mask);
 }
+EXPORT_SYMBOL_GPL(security_file_permission);
 
 int security_file_alloc(struct file *file)
 {
